src/server/database_provider.py

```diff
@@ -0,0 +1,24 @@
+import sqlite3
+import os
+
+from constants import PATHS
+
+def getDatabaseConnection():
```

Comment:
Generally global public functions are considered bad practice in larger than single-file projects, I suggest enclosing this function and the others below in a class with `@classmethod` decorator.

Also, please fix linter warnings, that means docstrings, typehints, all the usual stuff :).